hail/src/main/scala/is/hail/expr/ir/functions/GenotypeFunctions.scala

```diff
@@ -19,19 +19,22 @@ object GenotypeFunctions extends RegistryFunctions {
         val i = cb.newLocal[Int]("i", 0)
 
         cb.whileLoop(i < pl.loadLength(), {
-          val iec = pl.loadElement(cb, i)
-          cb.define(iec.Lmissing)
-          cb += Code._fatal[Unit]("PL cannot have missing elements.")
-          cb.define(iec.Lpresent)
-          val pli = cb.newLocal[Int]("pli", iec.pc.tcode[Int])
-          cb.ifx(pli < m, {
-            cb.assign(m2, m)
-            cb.assign(m, pli)
-          }, {
-            cb.ifx(pli < m2,
-              cb.assign(m2, pli))
-          })
-          cb.assign(i, i + 1)
+          pl.loadElement(cb, i)
+            .consume(cb,
+              {
+                cb += Code._fatal[Unit]("PL cannot have missing elements.")
```

Comment:
The failing tests are because this branch of the consume ends in a throw, but then `consume` tries to add a goto statement after that. I recently added some checks to code builder to complain about adding statements after a control command like a goto or throw, after John several times hit hard to debug segfaults caused by such bugs.

I would fix `consume` by changing the `goto` line to `if (cb.isOpenEnded) cb.goto(Lafter)`. Alternatively, instead of using `consume` you could first `handle` the `iec`, to get the value out.

Reply:
ah, yeah, got it. Fixed to use handle, that's a good solution